content_handler/app.cc

```diff
@@ -123,24 +123,35 @@ void App::StartApplication(
     app::ApplicationPackagePtr application,
     app::ApplicationStartupInfoPtr startup_info,
     fidl::InterfaceRequest<app::ApplicationController> controller) {
-  // Name this process after the url of the application being launched.
-  std::string label =
-      "flutter:" + GetLabelFromURL(startup_info->launch_info->url);
-  mx::process::self().set_property(MX_PROP_NAME, label.c_str(), label.size());
+  if (controllers_.empty()) {
+    // Name this process after the url of the first application being launched.
+    base_label_ = "flutter:" + GetLabelFromURL(startup_info->launch_info->url);
+  }
 
   std::unique_ptr<ApplicationControllerImpl> impl =
       std::make_unique<ApplicationControllerImpl>(this, std::move(application),
                                                   std::move(startup_info),
                                                   std::move(controller));
   ApplicationControllerImpl* key = impl.get();
   controllers_.emplace(key, std::move(impl));
+
+  UpdateProcessLabel();
 }
 
 void App::Destroy(ApplicationControllerImpl* controller) {
   auto it = controllers_.find(controller);
   if (it == controllers_.end())
     return;
   controllers_.erase(it);
+  UpdateProcessLabel();
+}
+
+void App::UpdateProcessLabel() {
+  std::string label = base_label_;
+  if (controllers_.size() >= 2) {
+    label += " (+" + std::to_string(controllers_.size() - 1) + ")";
+  }
+  mx::process::self().set_property(MX_PROP_NAME, label.c_str(), label.size());
```

Comment:
The kernel will truncate the name at MX_MAX_NAME_LEN - 1. Maybe there should be logic to truncate the base_label_ rather that losing the number of additional apps?